modules/apps/frontend-js/frontend-js-test/src/testFunctional/tests/WalkthroughPopover.testcase

```diff
@@ -61,6 +61,26 @@ definition {
 		}
 	}
 
+	@description = "LPS-154644. Assert popover mode can display step number."
+	@priority = "4"
+	test CanDisplayStepNumber {
+		task ("And given click hotspot element") {
+			Click(locator1 = "Walkthrough#HOTSPOT");
+		}
+
+		task ("When click through all primary step buttons") {
+			for (var number : list "2,3,4,5") {
+				Click(locator1 = "Walkthrough#NEXT_POPOVER");
+
+				task ("Then step number is displayed consecutively") {
```

Comment:
Would it be better to Click(ok) > Assert(step 2) > Click(ok) > Assert(step 3) > etc... or is looping ok? 

cc: @john-co

Reply:
I'm ok with this loop as an exception. I know we wanted to avoid loops to make poshi as readable as possible...but I also think for this specific test it's not too complicated.